src/main/java/HttpRequest.java

```diff
@@ -0,0 +1,38 @@
+package com.td.HttpServer;
+
+public class HttpRequest implements HttpProtocal {
+
+  private String method;
+  private String path;
+  private String version;
+
+  public HttpRequest(String request) {
```

Comment:
One benefit of having an `HttpRequest` object is that any other objects working with requests can accept an `HttpRequest` and just call methods on it like `getMethod`, `getPath`, etc. How the `HttpRequest` gets _created_ is a separate issue, and is probably isolated to two general places: 1) something in your production code responsible for assembling the request based on the parsed input received by the server, and 2) your tests (where you need to make requests).

My point here is that the request object doesn't need to be responsible for "assembling itself" from a string. Rather, one object (maybe a `RequestBuilder`?) can know the details of creating a request, and from that point on you have a convenient object to work with.

How dead-simple could you make the `HttpRequest`?